internal/test/e2e/artifacts.go

```diff
@@ -0,0 +1,67 @@
+package e2e
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/aws/eks-anywhere/internal/pkg/s3"
+	"github.com/aws/eks-anywhere/internal/pkg/ssm"
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+func (e *E2ESession) uploadGeneratedFilesFromInstance(testName string) {
+	logger.V(1).Info("Uploading log files to s3 bucket")
+	command := fmt.Sprintf("aws s3 cp /home/e2e/%s/ %s/%s/ --recursive",
+		e.instanceId, e.generatedArtifactsBucketPath(), testName)
+
+	if err := ssm.Run(e.session, e.instanceId, command); err != nil {
+		logger.Error(err, "error uploading log files from instance")
+	} else {
+		logger.V(1).Info("Successfully uploaded log files to S3")
+	}
+}
+
+func (e *E2ESession) uploadDiagnosticArchiveFromInstance(testName string) {
+	bundleNameFormat := "support-bundle-*.tar.gz"
+	logger.V(1).Info("Uploading diagnostic bundle to s3 bucket")
+	command := fmt.Sprintf("aws s3 cp /home/e2e/ %s/%s/ --recursive --exclude \"*\" --include \"%s\"",
```

Comment:
Is it possible to make `/home/e2e` a string variable? We have a lot of references to this

Reply:
yeah. I'll do it in a separate PR to avoid mixing more refactors with this functionality

Reply:
I have the changes ready, just need to merge this one first (I built it on top)